### PR TITLE
[cli][start] Added development session module

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- Split out of https://github.com/expo/expo/pull/16517

# How

- Added module for starting the development session API with timer / keep alive code.

# Test Plan

- Added unit tests.

### CODE REVIEW

File: packages/expo/cli/start/server/DevelopmentSession.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    this.stop();
+
+    this.timeout = setTimeout(() => this.startAsync({ exp, runtime }), UPDATE_FREQUENCY);
+  }
+
+  /** Get all recent devices for the project. */
+  private async getDeviceInstallationIdsAsync(): Promise<string[]> {
+    const { devices } = await ProjectDevices.getDevicesInfoAsync(this.projectRoot);
+    return devices?.map(({ installationId }) => installationId);
```
Comment (ide) [inline packages/expo/cli/start/server/DevelopmentSession.ts:69]: This is typed as nonnull, plus the return type of this function itself is nonnull
```suggestion
    return devices.map(({ installationId }) => installationId);
```

File: packages/expo/cli/start/server/DevelopmentSession.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  /** Stop notifying the Expo servers that the development session is running. */
+  public stop() {
+    clearTimeout(this.timeout);
```
Comment (ide) [inline packages/expo/cli/start/server/DevelopmentSession.ts:74]:
```suggestion
    clearTimeout(this.timeout);
    this.timeout = null;
```

**Review by ide — APPROVED** [review]
Requesting minor changes before merging.

File: packages/expo/package.json [hunk 3]
```diff
@@ -71,6 +71,9 @@
     "@expo/rudder-sdk-node": "1.1.1",
     "@expo/spawn-async": "1.5.0",
     "@expo/vector-icons": "^12.0.4",
+    "@expo/dev-server": "0.1.103",
+    "@expo/devcert": "^1.0.0",
+    "@expo/webpack-config": "0.16.16",
```
[the remark this answers is not in the document]
Comment (EvanBacon) [reply packages/expo/package.json:76]: I'll pull it from this PR but we will need to think of something lol. Considering moving the relevant code into expo/webpack-config then having the package locally installed as a requirement to run web.

File: packages/expo/cli/start/server/DevelopmentSession.ts (hunk 1 above)
Comment (EvanBacon) [reply packages/expo/cli/start/server/DevelopmentSession.ts:69]: I've noticed the TS in this repo (or at least the CLI folder) doesn't account for null. For example:

```ts
const foo: number | null = null;
```
if you hover `foo` in `expo/cli` you get `const foo: number` if you do the same in `expo-cli` you get `const foo: null`.

Not sure if you know how to fix this.

Comment (ide) [reply packages/expo/cli/start/server/DevelopmentSession.ts]: It is probably a tsconfig.json issue where strictNullChecks is not enabled: https://www.typescriptlang.org/tsconfig#strictNullChecks. If tsc isn't loading our tsconfig.json and is using the default rules, that could be the issue.

**Pull request merged**